src/iop/zonesystem.c

```diff
@@ -203,7 +203,7 @@ static void process_common_cleanup(struct dt_iop_module_t *self, dt_dev_pixelpip
   const int ch = piece->colors;
   const int size = d->params.size;
 
-  if(piece->pipe->mask_display) dt_iop_alpha_copy(ivoid, ovoid, width, height);
+  if(piece->pipe->mask_display & DT_DEV_PIXELPIPE_DISPLAY_MASK) dt_iop_alpha_copy(ivoid, ovoid, width, height);
```

Comment:
If it were only process() and process_sse2() I'd agree to put this into a central place. However, we also have process_cl(). If we would want a central dt_iop_alpha_copy_cl() this function would require one temporary buffer in addition to input and output (copy output to tmp, then copy xyz from tmp and w from input to output). This would put an additional requirement on tiling (memory is short on the GPU): not very good.

Ideally (and that is how things are handled in all OpenCL modules) each module takes care by itself to transfer the alpha channel with the buffers that it anyhow has reserved. This is the most efficient way in OpenCL.

Now this would also be the ideal case in process() and process_sse2(). But this tends to be forgotten by the module authors sometimes. Then typically we need to fix it (e.g. see ticket #11601) and I normally don't want to interfere too much with the original code. That's the reason why dt_iop_alpha_copy() has been added.